### PR TITLE
DO_deficit function and tests

### PR DESCRIPTION
**Pull request by jread-usgs**

### CODE REVIEW

**Comment by jread-usgs** [comment]
@aappling-usgs want to take a peek and provide feedback? This is for #15 and I will start working on #17

File: R/o2_at_sat.R [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#' @importFrom unitted u v get_units 
+#' @keywords internal
+o2_at_sat <- function(temp.water, pressure.air, salinity.water = u(0,'PSU'), ...){
+  
+  units <- sapply(list(temp.water, pressure.air, salinity.water), get_units)
+  
+  with.units <- TRUE
+  
+  if (any(is.na(units)) | !all(units == c("degC", "mb", "PSU"))){
+    with.units <- FALSE
```
Comment (aappling-usgs) [inline R/o2_at_sat.R]: So this call passes without error:
```
o2_at_sat(temp=u(77,"degF"), press=u(1.1,"atm"), sal=0)
```
is that the desired behavior? I think I'd rather see that fail.

Comment (jread-usgs) [inline R/o2_at_sat.R]: no, that should fail. ugg. will fix and tack on to this PR

Comment (aappling-usgs) [inline R/o2_at_sat.R]: ok. other than that looks great

**Comment by jread-usgs** [comment]
Hey @rBatt, we are adding units! Join the party!

**Comment by rBatt** [comment]
@jread-usgs What the theme of the party?

**Comment by aappling-usgs** [comment]
looking good!

**Comment by jread-usgs** [comment]
@rBatt party theme: `LakeMetabolizer + streamMetabolizer = BFFs`

**Pull request merged**